src/st/class-wpml-pb-integration.php

```diff
@@ -195,18 +195,60 @@ public function do_shutdown_action() {
 
 	public function new_translation( $translated_string_id ) {
 		if ( ! $this->is_registering_string ) {
-			foreach ( $this->strategies as $strategy ) {
+			$this->with_strategies( function( $strategy ) use ( $translated_string_id ) {
 				$this->factory->get_string_translations( $strategy )->new_translation( $translated_string_id );
-			}
+			} );
 			$this->new_translations_recieved = true;
 		}
 	}
 
+	/**
+	 * @param callable $callable
+	 * 
+	 * @return mixed
+	 */
+	private function with_strategies( callable $callable ) {
+		foreach ( $this->strategies as $strategy ) {
+			return $callable( $strategy );
```

Comment:
@strategio This doesn't look right. This code will only ever call the `$callable` once with the first strategy.

Reply:
How could I miss it @brucepearson! Thanks for catching it :)

I also strenghen the test with 3 added strategies, and updated the code with inversion of functions nesting (`with_strategies` is now inside the mapping function).